tsl/src/bgw_policy/job_api.c

```diff
@@ -201,10 +201,20 @@ static BgwJob *
 find_job(int32 job_id, bool null_job_id, bool missing_ok)
 {
 	BgwJob *job;
+	bool got_lock;
+	LOCKTAG tag;
 
 	if (null_job_id && !missing_ok)
 		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE), errmsg("job ID cannot be NULL")));
 
+	/* take advisory lock before relation lock */
+	if (!(got_lock = ts_lock_job_id(job_id,
+									RowShareLock,
+									/* session_lock */ false,
+									&tag,
+									/* block */ true)))
+		return NULL;
+
 	job = ts_bgw_job_find(job_id, CurrentMemoryContext, !missing_ok);
```

Comment:
Is there a particular reason (except that the parameter `missing_ok` is currently not supported) why we don't use `ts_bgw_job_find_with_lock` here? If we would use this function, we would not have to take care of the advisory lock ourselves.

Reply:
It causes strange failures and I wanted to investigate as a follow-up PR. This is duplicate code, so I think that we should use the same function for both cases.